.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,26 @@
+name: Static Analysis
+
+on: pull_request
```

Comment:
The Azure pipeline is configured to also run nightly. Wouldn't there be no results for the master branch (which you link to in the last commit) if this is only run for pull requests?

Reply:
I'm not sure. I assume "pull_request" builds are not run on the PR branch, but on the merge of PR and target branch. And if so, GitHub could be smart enough to associate the result of such runs also to the commit after the real merge happened (if nothing else was merged in between).

Anyway, as static analysis does not happen in the Azure pipeline anymore, there also are no nightly runs of detekt.

Reply:
> I assume "pull_request" builds are not run on the PR branch, but on the merge of PR and target branch.

That would surprise me, it could make interpretation of results very difficult. I also don't see the value, because the merge at the time the workflows are triggered is not necessarily the same as the result when the PR is actually merged.
Anyway, I'm fine with testing this with the static analysis job, but I expect that we will have to configure it to run on the master branch. The test jobs should still be triggered nightly, so this one could be triggered nightly as well, or, because it's cheap, for every push to master.
If you plan to do more test with GitHub actions, this looks like a useful feature we should try:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-ignoring-paths